pkg/event_handler/action_triggered_handler.go

```diff
@@ -3,94 +3,66 @@ package event_handler
 import (
 	"errors"
 	"fmt"
-	cloudevents "github.com/cloudevents/sdk-go/v2"
-	"github.com/keptn/go-utils/pkg/lib/keptn"
 	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
+	"github.com/keptn/keptn/go-sdk/pkg/sdk"
 	"net/http"
-	"net/url"
 	"os"
 	"strconv"
 )
 
 const ActionToggleFeature = "toggle-feature"
 
 type ActionTriggeredHandler struct {
-	Logger keptn.LoggerInterface
-	Event  cloudevents.Event
 }
 
-func (eh ActionTriggeredHandler) HandleEvent() error {
-	var shkeptncontext string
-	_ = eh.Event.Context.ExtensionAs("shkeptncontext", &shkeptncontext)
+// NewActionTriggeredHandler creates a new ActionTriggeredHandler
+func NewActionTriggeredHandler() *ActionTriggeredHandler {
+	return &ActionTriggeredHandler{}
+}
 
+// Execute handles the incoming cloud events
+func (g *ActionTriggeredHandler) Execute(k sdk.IKeptn, event sdk.KeptnEvent) (interface{}, *sdk.Error) {
 	actionTriggeredEvent := &keptnv2.ActionTriggeredEventData{}
 
-	err := eh.Event.DataAs(actionTriggeredEvent)
-	if err != nil {
-		eh.Logger.Error("feature toggle remediation action not well formed: " + err.Error())
-		return errors.New("feature toggle remediation action not well formed: " + err.Error())
+	if err := keptnv2.Decode(event.Data, actionTriggeredEvent); err != nil {
+		return nil, &sdk.Error{Err: err, StatusType: keptnv2.StatusErrored, ResultType: keptnv2.ResultFailed, Message: "feature toggle remediation action not well formed: " + err.Error()}
```

Comment:
I think this error message is misleading. At this point we are just trying to decode an `action.triggered` event, therefore the error message could be `"failed to decode action.triggered event: " + err.Error()`.